### ISSUE TITLE
Style/SymbolProc and Style/AutoResourceCleanup do not place nicely together

### ISSUE THREAD

**Issue by amilligan** [bbatsov/rubocop#2248]
Specifically, when using File.open.  With this code:

```ruby
File.open(path) { |f| f.read }
```

the Style/SymbolProc cop fails thusly:

```bash
C: Style/SymbolProc: Pass &:read as an argument to open instead of a block.
    File.open(path) { |f| f.read }
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

However with the suggested code:

```ruby
File.open(path, &:read)
```

the Style/AutoResourceCleanup cop fails:

```bash
C: Style/AutoResourceCleanup: Use the block version of File.open.
    File.open(path, &:read)
    ^^^^^^^^^^^^^^^^^^^^^^^
```

I believe this is due to the change to Style/SymbolProc after which that cop now applies to methods that take parameters.
